client/recordUtil.js

```diff
@@ -262,6 +273,12 @@ const mergeRecords = (recordsAndObjects) => {
  */
 module.exports.resolveRecords = (recordsAndExistingObjects) => {
   let resolvedRecords = []
+  recordsAndExistingObjects.forEach((item) => {
+    if (item) {
```

Comment:
is the `item` truthiness check necessary? in `mergeRecords()` we do a similar `forEach()` while assuming the values of item[0] and item[1].